graph/derive/src/lib.rs

```diff
@@ -0,0 +1,27 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{parse_macro_input, DeriveInput};
+
+#[proc_macro_derive(CheapClone)]
+pub fn derive_cheap_clone(input: TokenStream) -> TokenStream {
+    // Parse the input tokens into a syntax tree
+    let input = parse_macro_input!(input as DeriveInput);
+    let crate_name = std::env::var("CARGO_PKG_NAME").unwrap();
+
+    let cheap_clone = if crate_name == "graph" {
+        quote! { crate::cheap_clone::CheapClone }
+    } else {
+        quote! { graph::cheap_clone::CheapClone }
+    };
+
+    let name = input.ident;
+    let generics = input.generics;
+    // Build the output, possibly using the input
+    let expanded = quote! {
+        // The generated impl
+        impl #generics #cheap_clone for #name #generics { }
```

Comment:
That's true for impls that were diligent and destructured `self` - spoiler, I've just stuck a `impl CacheWeight for Type {}` into a lot of places where you have the same problem.

The 'right' way to derive this will also need to add bounds on generics; like for `Type<T>` it'll need to generate `impl<T: CheapClone> CacheWeight for Type<T>` .. ugh

Reply:
Yhea even the built-in derives haven't figured out the bounds. Happy to proceed however you prefer so approving.